Add steps for self-hosted environments (#8)
MicrosoftDocs/vsonline: vs-online/how-to/vscode.md

### Self-hosted

In addition to cloud-hosted environments, you can also "bring your own" self-hosted environments, registering them with VS Online.

> On the host machine, [install VS Code and the VS Online extension](#install).

Reviewer: I intentionally don't use the word "host" because that's not a concept that's been introduced anywhere in the docs.

We either need to explain what that means, or try to move this part of the text back to the original explanation:
```suggestion
To do so, first [install VS Code and the VS Online extension](#install) on the environment you'd like to register. Then use the **VS Online: Register Local Environment** command in the [command palette], and follow the prompts:
```

Author: Ok, changing "host machine" to "machine you wish to register as a self-hosted environment". Let me know if you had other thoughts in mind :)

Author: Nevermind, I am just going to revert this change because I figured out your comment below.

> Use the **VS Online: Register Local Environment** command from the [command palette](https://code.visualstudio.com/docs/getstarted/userinterface#_command-palette), and follow the prompts:

Reviewer: If you accept the above suggested edit, this line can be deleted.

Author: Got it - done.

> - If no folders are currently open, you will be prompted to select one. This folder will be opened every time you connect to this environment from another machine. However, you can open any folder on the machine after connecting.

Reviewer: ```suggestion
- If no folders are currently open in VS Code, you will be prompted to select one. The selected folder will be automatically opened every time you connect to the created environment from another machine. However, you can still change to any folder on the machine after connecting.
```

Author: Looks good! Added.

> ![Register a static environment in Visual Studio Code](../images/register-local-env-vsc-01.png)

Reviewer: There's no such thing as a "static environment" - that's only an internal name. 😉
```suggestion
![Register a Local Environment in Visual Studio Code](../images/register-local-env-vsc-01.png)
```

Author: Ah, yes, you're right. My bad! I even had it as local env in the image file name haha.

> - If no plan is selected, you will be prompted to select or create a plan. No charge is incurred for Self-hosted environments.

Reviewer: ```suggestion
- If no plan is selected, you will be prompted to select or create a plan. No charge is incurred for self-hosted environments.
```

Author: Good eye! Done!

After registering, your self-hosted environment will appear under a new tree view in the **VS Online** panel of the **Remote Explorer** side bar.

> ![Register a static environment in Visual Studio Code](../images/register-local-env-vsc-02.png)

Reviewer: See above comment about the usage of the word "Static"

```suggestion
![Self-hosted Environments in Visual Studio Code](../images/register-local-env-vsc-02.png)
```

Author: Done!

> You can now connect from any machine with the VS Online extension installed. The first time you connect will take longer than usual.

Reviewer: ```suggestion
You can now connect from any machine with the VS Online extension installed. The first time you connect may take longer than usual.
```

Author: Done!

> If your self-hosted environment becomes unavailable for any reason, you can attempt to [restore](#restore-a-self-hosted-environment) it.

Reviewer: I recommend moving the "Restore" section to the troubleshooting document.  If you do that, then this sentence would be re-written to something like 

> If your self-hosted environment becomes unavailable for any reason, see our [troubleshooting](ACTUAL LINK GOES HERE) reference documentation.

Author: Sounds good - done!

<!-- TODO: Add context menu screenshot -->

## Delete a cloud-hosted environment

The actively connected environment cannot be deleted from within VS Code, however while [disconnected from an environment](#disconnect-from-an-environment), there's two ways to permanently delete it:

1. Use the **VS Online: Delete Environment** command in the [command palette](https://code.visualstudio.com/docs/getstarted/userinterface#_command-palette) to select the environment to be deleted then press the **Delete** button on the confirmation prompt.
2. Right-click the name of the disconnected environment in the **VS Online** panel to reveal a context menu with a **Delete Environment** option. Select it and press the **Delete** button on the confirmation prompt.

> ## Restore a self-hosted environment

Reviewer: Please move this "Restore" section to the troubleshooting document.

Author: Good call - done!
